client.go

```diff
@@ -252,16 +254,14 @@ func getTableUpdatesFromRawUnmarshal(raw map[string]map[string]RowUpdate) TableU
 
 func clearConnection(c *rpc2.Client) {
 	connectionsMutex.Lock()
+	defer connectionsMutex.Unlock()
 	delete(connections, c)
-	connectionsMutex.Unlock()
 }
 
 func handleDisconnectNotification(c *rpc2.Client) {
 	disconnected := c.DisconnectNotify()
-	select {
```

Comment:
This select was redundant. select makes sense when there are multiple possible cases. It  doesn't make sense to use select when there's a single possible case.

This changes the code to read once from the channel and perform that operation. The effect is the same.